File: LeetCode/14. Longest Common Prefix/elixir/solution.exs
defmodule Solution do
  @spec longest_common_prefix(strs :: [String.t()]) :: String.t()
  def longest_common_prefix(strs) do
    strs
    |> Enum.map(&String.to_charlist/1)
    |> Enum.zip_with(& &1)
    |> Enum.map(&Enum.uniq/1)
    |> Enum.reduce_while('', fn
      [a], acc -> {:cont, [a | acc]}
      _, acc -> {:halt, acc}
    end)
    |> Enum.reverse()
    |> to_string()
  end
end

File: LeetCode/14. Longest Common Prefix/java/Solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder answer = new StringBuilder();
        char c;
        for (int i = 0; i < strs[0].length(); i++) {
            c = strs[0].charAt(i);
            for (String str : strs) {
                if (str.length() < i + 1 || c != str.charAt(i)) {
                    return answer.toString();
                }
            }
            answer.append(c);
        }
        return answer.toString();
    }
}